### DISCUSSION TITLE
How to attach kuzu to undo/gdb

### DISCUSSION THREAD

**Discussion by yassersun** [kuzudb/kuzu#2274] [General]
Hi guys, I was trying to attach the process to gdb, but could not find any documentation related to that. I was hoping if you could share the steps/documentation for it?

I am running on Ubuntu 22.04 LTS, and have build the binary from source.

Thanks.

**Comment by acquamarin** [comment]
Are you using the c++ api or the kuzu shell?
If you are using c++ api:
In order to debug kuzu, you need to link your executable with the kuzu_static library. Then you can execute your executable with lldb/gdb. 
If you are using kuzu_shell:
Simply launch the shell with lldb/gdb.

BTW: You can join our [Discord ](https://kuzudb.com/chat) channel where you can ask questions and get response quickly.

**Reply by yassersun** [reply]
Hi @acquamarin , I have joined the slack channel you shared - just a quick question, how to load the symbol table for the shell method?
